app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/ActivityAnalysis.java

```diff
@@ -65,12 +81,56 @@ ActivityAmounts calculateActivityAmounts(List<? extends ActivitySample> samples)
                     previousAmount.addSeconds(sharedTimeDifference);
                     amount.addSeconds(sharedTimeDifference);
                 }
+
+                // add time
+                if (steps > 0 && sample.getKind() == ActivityKind.TYPE_ACTIVITY) {
+                    if (steps > maxSpeed) {
+                        maxSpeed = steps;
+                    }
+
+                    if (!stats.containsKey(steps)) {
+                        //System.out.println("Adding: " + steps);
+                        stats.put(steps, (long) 60);
```

Comment:
AFAIU this hardcodes 60 seconds for the time between two samples, right? We're trying to avoid such things because different devices may have different sample frequences. That's why we always calculate the time between two samples (see `timeDifference`).